src/components/Filter/FilterGroup.js

```diff
@@ -5,37 +5,54 @@ import FilterItem from './FilterItem';
 import styles from './FilterGroup.scss';
 import withStyles from 'isomorphic-style-loader/lib/withStyles';
 import {translateGroup, translateTag} from '../../util';
+import Collapse from 'react-bootstrap/lib/Collapse';
+import Glyphicon from 'react-bootstrap/lib/Glyphicon';
 
-const FilterGroup = ({groupKey, availableLessonsForTag, t, filterTags, onFilterCheck}) => {
-  const groupName = translateGroup(t, groupKey);
+export const FilterGroup = React.createClass({
+  getInitialState() {
+    return {
+      showFilterTags: true
+    };
+  },
+  changeState() {
+    this.setState({['showFilterTags']: !this.state['showFilterTags']});
+  },
+  render() {
+    const {groupKey, availableLessonsForTag, t, filterTags, onFilterCheck} = this.props;
+    const {showFilterTags} = this.state;
+    const groupName = translateGroup(t, groupKey);
+    if (groupName) {
+      const filterItems = Object.keys(filterTags).map((tagKey) => {
+        const onCheck = () => onFilterCheck(groupKey, tagKey);
+        const numberOfLessonsForTag = availableLessonsForTag[tagKey];
+        const tagName = translateTag(t, groupKey, tagKey);
 
-  if (groupName) {
-    const filterItems = Object.keys(filterTags).map((tagKey) => {
-      const onCheck = () => onFilterCheck(groupKey, tagKey);
-      const numberOfLessonsForTag = availableLessonsForTag[tagKey];
-      const tagName = translateTag(t, groupKey, tagKey);
-
-      return tagName ? (
-        <FilterItem
-          key={tagKey}
-          tagName={tagName}
-          numberOfLessons={numberOfLessonsForTag}
-          checked={filterTags[tagKey]}
-          onCheck={onCheck}
-        />
-      ) : null;
-    });
-
-    return (
-      <div className={styles.filterGroup}>
-        <h4>{groupName}</h4>
-        {filterItems}
-      </div>
-    );
-  } else {
-    return null;
+        return tagName ? (
+          <FilterItem
+            key={tagKey}
+            tagName={tagName}
+            numberOfLessons={numberOfLessonsForTag}
+            checked={filterTags[tagKey]}
+            onCheck={onCheck}
+          />
+        ) : null;
+      });
+      return (
+        <div className={styles.filterGroup}>
+          <h4 className={styles.name} onClick={() => this.changeState()}>
+            <Glyphicon className={styles.glyph} glyph={!showFilterTags ? 'chevron-right' : 'chevron-down'}/>{groupName}
+          </h4>
+          <Collapse in={showFilterTags}>
+            <div>{filterItems}</div>
+          </Collapse>
+        </div>
+      );
+    }
```

Comment:
Forslag til endringer:
* ikke wrappe `this.changeState`
* `{groupName}` på egen linje
* Legge til className `styles.filterItems` for å unngå margin collapse. Se f.eks. https://stackoverflow.com/questions/19718634/how-to-disable-margin-collapsing (du merker at listen "hopper" når du expander og collapser den hvis du ikke har denne med)

```
return (
        <div className={styles.filterGroup}>
          <h4 className={styles.name} onClick={this.changeState}>
            <Glyphicon className={styles.glyph} glyph={!showFilterTags ? 'chevron-right' : 'chevron-down'}/>
            {groupName}
          </h4>
          <Collapse in={showFilterTags}>
            <div className={styles.filterItems}>{filterItems}</div>
          </Collapse>
        </div>
      );
```